packages/styles/src/mixins/_forms.scss

```diff
@@ -257,37 +285,14 @@ $fd-input-field-height--compact: 1.625rem;
     )
   );
 
-  .#{$block}__addon {
-    &:first-child {
-      .#{$block}__button {
-        @include fd-set-border-right(var(--fdInputGroup_ControlButton_SideBorder));
+  @include _fd-input-group-button-first-child(".#{$fd-namespace}-button:first-child:not(.#{$fd-namespace}-input-group__button)");
+  @include _fd-input-group-button-last-child(".#{$fd-namespace}-button:not(:first-child, .#{$fd-namespace}-input-group__button)");
 
-        @include fd-hover() {
-          @include fd-set-border-right(var(--fdInputGroup_ControlButton_SideBorder_Active));
-        }
-
-        @include fd-active() {
-          @include fd-set-border-right(var(--fdInputGroup_ControlButton_SideBorder_Active));
-        }
-      }
-    }
-
-    &:not(:first-child) {
-      .#{$block}__button {
-        @include fd-set-border-left(var(--fdInputGroup_ControlButton_SideBorder));
-
-        @include fd-hover() {
-          @include fd-set-border-left(var(--fdInputGroup_ControlButton_SideBorder_Active));
-        }
-
-        @include fd-active() {
-          @include fd-set-border-left(var(--fdInputGroup_ControlButton_SideBorder_Active));
-        }
-      }
-    }
-  }
+  // To support `fd-input-group__addon` and avoid breaking changes
+  @include _fd-input-group-button-first-child(".#{$block}__addon:first-child .#{$fd-namespace}-button");
+  @include _fd-input-group-button-last-child(".#{$block}__addon:not(:first-child) .#{$fd-namespace}-button");
 
-  .#{$block}__button {
+  .#{$fd-namespace}-button {
```

Comment:
with this you are breaking the self-contained rule. You are directly modifying button inside Select component.